extractor/main_test.go

```diff
@@ -0,0 +1,405 @@
+
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, tx := range mockTx[:TxPerSourceFile] {
+
+			err := writeTxToFile(tx, file)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+		mockTx = mockTx[TxPerSourceFile:]
+	}
+
+	for i := 0; i < NumSourceFiles; i++ {
+		testFiles[i] = filepath.Join(tempDir, testFiles[i])
+	}
+
+	t.Cleanup(func() {
+		err := os.RemoveAll(tempDir)
+		if err != nil {
+			fmt.Printf("could not clean up temp dir, %v", err)
+			return
+		}
+	})
+
+	return testFiles
+}
+
+func generateMockMsgs(t *testing.T) ([]std.Msg, []vm.MsgAddPackage) {
+	t.Helper()
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	testAddresses := []string{
+		"g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5",
+		"g1f4v282mwyhu29afke4vq5r2xzcm6z3ftnugcnv",
+		"g127jydsh6cms3lrtdenydxsckh23a8d6emqcvfa",
+		"g1u7y667z64x2h7vc6fmpcprgey4ck233jaww9zq",
+		"g1rrf8s5mrmu00sx04fzfsvc399fklpeg2x0a7mz",
+	}
+
+	var ret []std.Msg
+	var addPkgRet []vm.MsgAddPackage
+
+	pkgID := 0
+
+	for i := 0; i < NumMsg; i++ {
+
+		var (
+			randNum          = int(r.Uint32())
+			msg              std.Msg
+			randAddressIndex = randNum % len(testAddresses)
+			callerAddr       = addressFromString(testAddresses[randAddressIndex])
+			deposit          = std.NewCoins(std.NewCoin("foo", int64(randNum%MaxDepositAmount+1)))
+			path             = "gno.land/"
+			pkgName          = "package" + strconv.Itoa(pkgID)
+		)
+
+		if randNum%2 == 0 {
+			path += "r/"
+		} else {
+			path += "p/"
+		}
+
+		switch randNum % 3 {
+
+		case 0: // Making vm.MsgAddPackage msg
+
+			var files []*std.MemFile
+
+			path += pkgName
+
+			for j := 0; j < randNum%MaxFilesPerPkg+1; j++ {
+				file := &std.MemFile{
+					Name: "t" + strconv.Itoa(j) + ".gno",
+					Body: randString(int(r.Uint32()) % MaxFileBodyLength),
+				}
+				files = append(files, file)
+			}
+
+			msg = vm.MsgAddPackage{
+				Creator: callerAddr,
+				Package: &std.MemPackage{
+					Name:  pkgName,
+					Path:  path,
+					Files: files,
+				},
+				Deposit: deposit,
+			}
+			addPkgRet = append(addPkgRet, msg.(vm.MsgAddPackage))
+			break
+		case 1: // Making vm.MsgCall msg
+
+			args := make([]string, MaxArgs-randNum%2)
+			for i := range args {
+				args[i] = randString(10)
+			}
+
+			msg = vm.MsgCall{
+				Caller:  callerAddr,
+				Send:    deposit,
+				PkgPath: path + pkgName,
+				Func:    "Func" + strconv.Itoa(i),
+				Args:    args,
+			}
+			break
+		case 2: // Making bank.MsgSend
+
+			// Remove already used address
+			ta := append(testAddresses[:randAddressIndex], testAddresses[randAddressIndex+1:]...)
+
+			msg = bank.MsgSend{
+				FromAddress: callerAddr,
+				ToAddress:   addressFromString(ta[randNum%len(ta)]),
+				Amount:      deposit,
+			}
+		}
+		ret = append(ret, msg)
+		pkgID++
+	}
+
+	return ret, addPkgRet
+}
+
+func addressFromString(addr string) crypto.Address {
+	ret, err := crypto.AddressFromString(addr)
+	if err != nil {
+		fmt.Errorf("cannot convert string to address, %v", err)
```

Comment:
(I discovered that no linter checks for this, so I made a [PR to the best candidate](https://github.com/kisielk/errcheck/pull/235) :))